src/SetHo.hs

```diff
@@ -194,6 +204,9 @@ runSetter mconfig rootName initialValue userPollForNewMessage sendRequest userCo
 
   _ <- on buttonDiff Gtk.buttonActivated printDiff
 
+  _ <- on buttonRevertToDefaults Gtk.buttonActivated
+       ((getLatestStaged >>= revertToDefaults) *> takeLatestUpstream)
```

Comment:
I haven't looked at this code in a while, but I don't immediately understand this. Can you add a short comment on the use case? Specifically:
* why do you need the latest staged DTree in order to revert to defaults?
* why do you call takeLatestUpstream afterwards? in a GCS use-case the userRevertToDefaults won't have an affect by the time takeLatestUpstream is called, is that right?

Reply:
The reason to call `takeLatestUpstream` was so that reverting to defaults in the upstream system would also revert the staged values.  It appears you are correct that this doesn't work as I'd hoped.  I've deleted it; the user will just have to revert the staged values with a separate button press.